definitions/checks/foreman/puppet_class_duplicates.rb

```diff
@@ -0,0 +1,48 @@
+module Checks
+  module Foreman
+    class PuppetClassDuplicates < ForemanMaintain::Check
+      metadata do
+        label :puppet_class_duplicates
+        for_feature :foreman_database
+        description 'Check for duplicate Puppet class records'
+        tags :pre_upgrade
+        confine do
+          feature(:foreman_database) && feature(:foreman_database).check_max_version('foreman', '1.20')
+        end
```

Comment:
I added a confine to restrict the checks only for version 1.20 and lower.

Reply:
The `check_max_version` should be available directly in current context., no need to call it on `feautre(:foreman_database)`. Also, I guess the check should be only on `1.19` and below (provided the fix will be in `1.20`?

Reply:
On the other hand, it probably doesn't hurt, if we had is on 1.20 as well